demo/admin/src/news/handmade/NewsForm.tsx

```diff
@@ -0,0 +1,337 @@
+        if (await saveConflict.checkForConflicts()) {
+            throw new Error("Conflicts detected");
+        }
+
+        const output = {
+            ...state,
+
+            image: rootBlocks.image.state2Output(state.image),
+            content: rootBlocks.content.state2Output(state.content),
+        };
+
+        if (mode === "edit") {
+            if (!id) {
+                throw new Error("Missing id in edit mode");
+            }
+            await client.mutate<GQLUpdateNewsMutation, GQLUpdateNewsMutationVariables>({
+                mutation: updateNewsMutation,
+                variables: { id, input: output },
+            });
+        } else {
+            const { data: mutationResponse } = await client.mutate<GQLCreateNewsMutation, GQLCreateNewsMutationVariables>({
+                mutation: createNewsMutation,
+                variables: { scope, input: output },
+            });
+            if (!event.navigatingBack) {
+                const id = mutationResponse?.createNews.id;
+                if (id) {
+                    setTimeout(() => {
+                        stackSwitchApi.activatePage("edit", id);
+                    });
+                }
+            }
+        }
+    };
+
+    if (error) throw error;
+
+    if (loading) {
+        return <Loading behavior="fillPageHeight" />;
+    }
+
+    return (
+        <FinalForm<FormValues> apiRef={formApiRef} onSubmit={handleSubmit} mode={mode} initialValues={initialValues}>
+            {({ values, form }) => (
+                <EditPageLayout>
+                    {saveConflict.dialogs}
+                    <Toolbar>
+                        <ToolbarItem>
+                            <IconButton onClick={stackApi?.goBack}>
+                                <ArrowLeft />
+                            </IconButton>
+                        </ToolbarItem>
+                        <ToolbarTitleItem>
+                            <FormattedMessage id="news.News" defaultMessage="News" />
+                        </ToolbarTitleItem>
+                        <ToolbarFillSpace />
+                        <ToolbarActions>
+                            <FinalFormSaveSplitButton hasConflict={saveConflict.hasConflict} />
+                        </ToolbarActions>
+                    </Toolbar>
+                    <MainContent>
+                        <BlockPreviewWithTabs
+                            previewUrl={`${siteConfig.blockPreviewBaseUrl}/news`}
+                            previewState={{
+                                ...values,
+                                image: DamImageBlock.createPreviewState(values.image, previewContext),
+                                content: NewsContentBlock.createPreviewState(values.content, {
+                                    ...previewContext,
+                                    parentUrl: `${match.url}/content`,
+                                }),
+                            }}
```

Comment:
Is it okay to create this on every render? Otherwise we'd need a form spy and store it in a state

Reply:
I'd say that's okay as long as it doesn't cause rendering performance issues. But I think @nsams can judge this better